server/services/order.js

```diff
@@ -0,0 +1,82 @@
+module.exports = class OrderService {
+    static ORDER_BUILD_AND_SEND_CARRIER = "ORDER_BUILD_AND_SEND_CARRIER";
+
+    constructor(carrierService, waypointService, starService, starUpgradeService) {
+        this.carrierService = carrierService;
+        this.waypointService = waypointService;
+        this.starService = starService;
+        this.starUpgradeService = starUpgradeService;
+        this.orderHandlers = {
+            ORDER_BUILD_AND_SEND_CARRIER: this._handleBuildAndSendCarrier.bind(this)
+        }
+    }
+
+    async processOrdersForPlayer(game, player) {
+        if (!player.scheduledOrders) {
+            player.scheduledOrders = [];
+        }
+
+        const newOrders = [];
+
+        for (let order of player.scheduledOrders) {
+            let success = false;
+            try {
+                success = await this._performOrder(game, player, order.orderType, order.data);
+            } finally {
+                if (!success) {
```

Comment:
I don't think you need to check `success` here, just do the stuff inside the if regardless.

Reply:
If no ships are at the star at the moment, no carrier can be built there, that's why there's a success flag. I should probably check the other star as well, but it allows retrying later.